cloudfoundry/resource_cf_isolation_segment_test.go

```diff
@@ -0,0 +1,165 @@
+	"fmt"
+	"testing"
+
+	"code.cloudfoundry.org/cli/cf/errors"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/terraform-providers/terraform-provider-cf/cloudfoundry/cfapi"
+)
+
+const segmentResource = `
+resource "cloudfoundry_org" "iso-org1" {
+	name = "iso-organization-one"
+}
+resource "cloudfoundry_org" "iso-org2" {
+	name = "iso-organization-two"
+}
+resource "cloudfoundry_isolation_segment" "segment1" {
+	name = "segment-one"
+}
+resource "cloudfoundry_isolation_segment_entitlement" "segment1_orgs" {
+	segment = "${cloudfoundry_isolation_segment.segment1.id}"
+	orgs = [
+    "${cloudfoundry_org.iso-org1.id}",
+    "${cloudfoundry_org.iso-org2.id}"
+  ]
+}
+`
+
+const segmentResourceUpdateName = `
+resource "cloudfoundry_org" "iso-org1" {
+	name = "iso-organization-one"
+}
+resource "cloudfoundry_org" "iso-org2" {
+	name = "iso-organization-two"
+}
+resource "cloudfoundry_isolation_segment" "segment1" {
+	name = "segment-one-name"
+}
+resource "cloudfoundry_isolation_segment_entitlement" "segment1_orgs" {
+	segment = "${cloudfoundry_isolation_segment.segment1.id}"
+	orgs = [
+    "${cloudfoundry_org.iso-org1.id}",
+    "${cloudfoundry_org.iso-org2.id}"
+  ]
+}
+`
+
+const segmentResourceUpdateOrgs = `
+resource "cloudfoundry_org" "iso-org1" {
+	name = "iso-organization-one"
+}
+resource "cloudfoundry_isolation_segment" "segment1" {
+	name = "segment-one-name"
+}
+resource "cloudfoundry_isolation_segment_entitlement" "segment1_orgs" {
+	segment = "${cloudfoundry_isolation_segment.segment1.id}"
+	orgs = [
+    "${cloudfoundry_org.iso-org1.id}"
+  ]
+}
+`
+
+func TestAccSegment_normal(t *testing.T) {
+	segRef := "cloudfoundry_isolation_segment.segment1"
+	entitleRef := "cloudfoundry_isolation_segment_entitlement.segment1_orgs"
+
+	resource.Test(t,
+		resource.TestCase{
+			PreCheck:     func() { testAccPreCheck(t) },
+			Providers:    testAccProviders,
+			CheckDestroy: testAccCheckSegmentDestroyed("segment-one-name"),
+			Steps: []resource.TestStep{
+				resource.TestStep{
+					Config: segmentResource,
+					Check: resource.ComposeTestCheckFunc(
+						testAccCheckSegmentExists(segRef, entitleRef),
+						resource.TestCheckResourceAttr(segRef, "name", "segment-one"),
+						resource.TestCheckResourceAttr(entitleRef, "orgs.#", "2"),
+					),
+				},
+				resource.TestStep{
+					Config: segmentResourceUpdateName,
+					Check: resource.ComposeTestCheckFunc(
+						testAccCheckSegmentExists(segRef, entitleRef),
+						resource.TestCheckResourceAttr(segRef, "name", "segment-one-name"),
+						resource.TestCheckResourceAttr(entitleRef, "orgs.#", "2"),
+					),
+				},
+				resource.TestStep{
+					Config: segmentResourceUpdateOrgs,
+					Check: resource.ComposeTestCheckFunc(
+						testAccCheckSegmentExists(segRef, entitleRef),
+						resource.TestCheckResourceAttr(segRef, "name", "segment-one-name"),
+						resource.TestCheckResourceAttr(entitleRef, "orgs.#", "1"),
+					),
```

Comment:
May be add coverage of resource import, similar to other imports, e.g. https://github.com/mevansam/terraform-provider-cf/blob/07283690098bbfa8fe5fbf4a8415ccdaba4b8933/cloudfoundry/import_cf_buildpack_test.go#L9